src/Linkify.jsx

```diff
@@ -13,19 +38,62 @@ class Linkify extends React.Component {
     component: React.PropTypes.any,
     properties: React.PropTypes.object,
     urlRegex: React.PropTypes.object,
-    emailRegex: React.PropTypes.object
+    emailRegex: React.PropTypes.object,
+    handlers: React.PropTypes.arrayOf(
+      React.PropTypes.shape({
+        prefix: React.PropTypes.string.isRequired,
+        validate: React.PropTypes.func.isRequired,
+        normalize: React.PropTypes.func.isRequired
+      })
+    ),
+    fuzzyLink: React.PropTypes.bool,
+    fuzzyIP: React.PropTypes.bool,
+    fuzzyEmail: React.PropTypes.bool
   }
 
   static defaultProps = {
     className: 'Linkify',
     component: 'a',
     properties: {},
+    handlers: []
+  }
+
+  componentDidMount() {
+    this.addCustomHandlers();
+  }
+
+  componentDidUpdate(nextProps) {
+    this.addCustomHandlers();
+  }
+
+  addCustomHandlers() {
+    const { handlers } = this.props;
+
+    this.linkify = this.linkify || new LinkifyIt();
```

Comment:
How about we move `this.linkify = new LinkifiyIt();` into `componentDidMount()` so that `this.linkify` will always be defined when calling `addCustomHandlers()`.